eugl/mndwi.py

```diff
@@ -0,0 +1,381 @@
+        swir_bname = "BAND-11"
+
+    # search for the green and swir band
+    green_path = [f for f in paths if (product in f and green_bname in f)]
+    swir_path = [f for f in paths if (product in f and swir_bname in f)]
+
+    if not green_path:
+        raise Exception(f'could not find "{product}"{green_bname} in {granule}')
+
+    if not swir_path:
+        raise Exception(f'could not find "{product}"{swir_bname} in {granule}')
+
+    return green_path[0], swir_path[0]
+
+
+def get_RGB_bands(platform_id, product, paths):
+    """
+    Get the RGB band paths from the H5 file.
+
+    platform_id : str
+        Platform name, e.g. LANDSAT_8, SENTINEL_2A, SENTINEL_2B
+
+    product : str
+        name of reflectance product from which the relevant
+        MNDWI bands are selected from
+
+    paths : list
+        list of paths to datasets contained in the wagl h5 file
+
+    Returns
+    -------
+    red_path : str
+        path to red band
+
+    green_path : str
+        path to green band
+
+    blue_path : str
+        path to blue band
+    """
+
+    # for sentinel-2, landsat-8 and WV2
+    red_bname = "BAND-4"
+    green_bname = "BAND-3"
+    blue_bname = "BAND-2"
+
+    if (platform_id == "LANDSAT_7") or (platform_id == "LANDSAT_5"):
+        red_bname = "BAND-3"
+        green_bname = "BAND-2"
+        blue_bname = "BAND-1"
+
+    # search for the RGB bands
+    red_path = [f for f in paths if (product in f and red_bname in f)]
+    green_path = [f for f in paths if (product in f and green_bname in f)]
+    blue_path = [f for f in paths if (product in f and blue_bname in f)]
+
+    if not red_path:
+        raise Exception(f'could not find "{product}"{red_bname}')
+
+    if not green_path:
+        raise Exception(f'could not find "{product}"{green_bname}')
+
+    if not blue_path:
+        raise Exception(f'could not find "{product}"{blue_bname}')
+
+    return red_path[0], green_path[0], blue_path[0]
+
+
+def seadas_style_scaling(band_im, sfactor):
+    """
+    Perform a (NASA-OBPG) SeaDAS style scaling on a single band.
+    When applied to the red, green and blue bands this scaling
+    generates very impressive RGB's
+
+    Parameters
+    ----------
+    band_im : numpy.ndarray
+        reflectance image at either the  band [nRows, nCols]
+
+    sfactor : float or int
+        The multiplicative factor used to convert to reflectances
+
+    Returns
+    -------
+    scaled_im : numpy.ndarray [dtype = np.uint8]
+        Scaled and enhanced band with dimensions of [nRows, nCols].
+
+    Raises
+    ------
+        * Exception if len(rgb_ix) != 3
+
+    """
+
+    # specify coefficient used in transformation:
+    c1 = 0.091935692
+    c2 = 0.61788
+    c3 = 10.0
+    c4 = -0.015
+
+    scaled_im = c1 + c2 * np.arctan(c3 * (band_im / sfactor + c4))
+    scaled_im[scaled_im < 0] = 0
+    scaled_im[scaled_im > 1] = 1
+    return np.array(255 * scaled_im, order="C", dtype=np.uint8)
+
+
+def mndwi(wagl_h5_file, granule, out_fname):
```

Comment:
This function call is fairly lengthy.

Ideally split it into different components.

The actual mndwi calculation ideally should be its own function, that takes the required arrays, applies the function and returns an array.

Reply:
`mndwi()` calls `compute_mndwi()` which computes the mndwi using the input green and swir arrays.